### DISCUSSION TITLE
Few things regarding Mania skin editing, Hit position/Judgement line and visual offset

### DISCUSSION THREAD

**Discussion by LukeConfait** [ppy/osu#18059] [Ideas]
Apologies for the weird wording of the title. I'm following up on a reply to a thread on visual offset but I was told and now realize this makes more sense as its own topic.

Ok so the background to the issue is that when I change skins in mania or change the scroll speed I will end up hitting early or late from average based on how these change the way I read the notes. This I believe is unrelated to any delay or should be evenly affected by it.

Some other games provide a visual offset setting to counteract this. Osu actually has a feature that does a similar thing already which is to adjust the 'HitPosition' in the skin .ini file up or down based on whether the player is hitting early or late. its a bit weird since the height is not adjusted in timing values and a bit of fiddling is needed to get it right. 

There are 2 other issues this causes:
The first is that this requires loading the skin and editing the .ini file over and over again which is made worse in osulazer as the skin folder isn't accessible directly so it just takes more time to do.
The second is that this messes up the visuals of the skin especially with Long notes like this
![image](https://user-images.githubusercontent.com/86744792/166296016-ba3c1aa4-abcd-4448-8500-e7c920da7bdc.png)
as the key images for the receptors are fixed based on the skin element in the skin folder. Editing these is beyond what most players would be reasonably expected to do.

I have a few suggestions for what could help.
One would be to allow the HitPosition to be edited in game perhaps in a time based unit which could be converted to height based on the scroll speed.
The other is to change the system so the HitPosition is attached to the receptor images and the whole playfield is adjusted up or down to set them at the right place without needing to edit skin elements. 

These would not require any new offset be implemented and shouldn't affect hitsounds as this is based on visuals and the first part can already be done in a way.
Another aspect of this issue is that some people hit 'on' the judgment line or 'above' the judgement line depending on the skin.

**Comment by peppy** [comment]
Have you tried turning on screen scaling and adjusting the gameplay size? ie. set the Y offset to 0 and adjust the height % down to move your judgement point upwards on the screen.

**Reply by LukeConfait** [reply]
I didn't even think to try those settings. I gave the screen scaling setting a go and found that this does indeed have the effect of adjusting the height of the judgement line, however this also shrinks the gameplay elements which is not ideal. It would be preferable to be able to shift the entire playfield up or down without scaling and having finer control than just percentages would also be helpful.

Since this isn't controlling for all variables its hard to gauge the effect on hitting early or late but anecdotally I would say adjusting the HitPosition in the skin is still more effective.

**Reply by peppy** [reply]
Yeah, we'll eventually get to exposing those fully.
